core/types/block.libevm.go

```diff
@@ -209,5 +158,16 @@ func (b *Body) hooks() BodyHooks {
 // having been registered.
 type NOOPBodyHooks struct{}
 
-func (NOOPBodyHooks) AppendRLPFields(rlp.EncoderBuffer, bool) error { return nil }
-func (NOOPBodyHooks) DecodeExtraRLPFields(*rlp.Stream) error        { return nil }
+// The RLP-related methods of [NOOPBodyHooks] make assumptions about the struct
+// fields and their order, which we lock in here as a change detector. If this
+// breaks then it MUST be updated and the RLP methods reviewed + new
+// backwards-compatibility tests added.
+var _ = &Body{[]*Transaction{}, []*Header{}, []*Withdrawal{}}
+
+func (NOOPBodyHooks) RLPFieldsForEncoding(b *Body) ([]any, []any) {
+	return []any{b.Transactions, b.Uncles}, []any{b.Withdrawals}
+}
+
+func (NOOPBodyHooks) RLPFieldPointersForDecoding(b *Body) ([]any, []any) {
```

Comment:
nit named returns would help readability

```suggestion
func (NOOPBodyHooks) RLPFieldsForEncoding(b *Body) (required, optional []any) {
	return []any{b.Transactions, b.Uncles}, []any{b.Withdrawals}
}

func (NOOPBodyHooks) RLPFieldPointersForDecoding(b *Body) (required, optional []any) {
```

Reply:
No longer necessary with the introduction of `rlp.OptionalFields`.